pkg/ddl/reorg.go

```diff
@@ -882,6 +883,17 @@ func getReorgInfo(ctx *ReorgContext, jobCtx *jobContext, rh *reorgHandler, job *
 			tb = tbl.(table.PhysicalTable)
 		}
 		if mergingTmpIdx {
+			for _, element := range elements {
+				if !bytes.Equal(element.TypeKey, meta.IndexElementKey) {
+					continue
+				}
+				// If has a global index in elements, need start process at `tblInfo.ID`
+				// because there are some temporary global indexes prefixed with table ID.
+				idxInfo := model.FindIndexInfoByID(tblInfo.Indices, element.ID)
+				if idxInfo.Global {
+					pid = tblInfo.ID
```

Comment:
Is there always only a single index/element with IndexElementKey here?
What happens for `alter table t add index idx_1(b), add unique index idx_2(c) global`, where the indexes have different columns?
I'm not very familiar with the ingest add index code, but for non-index it will add all indexes in one go, one partition at a time. Is this the same for the mergingTmpIdx?

Reply:
For multi-schema change, we will combine add indexes into one reorg job. So the elements will greatter than 1. See https://github.com/pingcap/tidb/pull/47135 for details.

Reply:
Ah, if there is any global index in elements, it will first do all global indexes by reading from the TableID, then it will do local indexes by looping over each partitionID and all the elements.

So do we also loop over all partitions even if there only are global indexes added?

Reply:
Updated, fixed in `getNextPartitionInfo` function